broker-core/src/test/java/io/zeebe/broker/workflow/TimerStartEventTest.java

```diff
@@ -0,0 +1,324 @@
+
+    // when
+    testClient.deploy(REPEATING_MODEL);
+    assertThat(RecordingExporter.timerRecords(TimerIntent.CREATED).limit(2).count()).isEqualTo(2);
+    brokerRule.getClock().addTime(Duration.ofSeconds(2));
+
+    assertThat(
+            RecordingExporter.workflowInstanceRecords(EVENT_ACTIVATED)
+                .withElementId("end_2")
+                .withBpmnProcessId("process")
+                .withVersion(2)
+                .exists())
+        .isTrue();
+  }
+
+  @Test
+  public void shouldReplaceTimerStartWithNoneStart() {
+    // when
+    testClient.deploy(REPEATING_MODEL);
+
+    assertThat(RecordingExporter.timerRecords(TimerIntent.CREATED).exists()).isTrue();
+    brokerRule.getClock().addTime(Duration.ofSeconds(1));
+
+    // then
+    assertThat(RecordingExporter.timerRecords(TimerIntent.TRIGGERED).exists()).isTrue();
+
+    // when
+    final BpmnModelInstance nonTimerModel =
+        Bpmn.createExecutableProcess("process").startEvent("start_4").endEvent("end_4").done();
+    testClient.deploy(nonTimerModel);
+
+    assertThat(RecordingExporter.deploymentRecords(DeploymentIntent.CREATED).limit(2).count())
+        .isEqualTo(2);
+    brokerRule.getClock().addTime(Duration.ofSeconds(2));
+
+    // then
+    assertThat(RecordingExporter.timerRecords(TimerIntent.CANCELED).exists()).isTrue();
+    assertThat(RecordingExporter.timerRecords(TimerIntent.TRIGGERED).exists()).isTrue();
+
+    final long workflowInstanceKey = testClient.createWorkflowInstance("process");
+
+    final WorkflowInstanceRecordValue lastRecord =
+        RecordingExporter.workflowInstanceRecords(EVENT_ACTIVATED)
+            .withElementId("end_4")
+            .getFirst()
+            .getValue();
+
+    Assertions.assertThat(lastRecord)
+        .hasVersion(2)
+        .hasBpmnProcessId("process")
+        .hasWorkflowInstanceKey(workflowInstanceKey);
+  }
+
+  @Test
+  public void shouldUpdateTimerPeriod() {
+    // when
+    long beginTime = brokerRule.getClock().getCurrentTimeInMillis();
+    testClient.deploy(THREE_SEC_MODEL);
+    assertThat(RecordingExporter.timerRecords(TimerIntent.CREATED).exists()).isTrue();
+    brokerRule.getClock().addTime(Duration.ofSeconds(3));
+
+    // then
+    TimerRecordValue timerRecord =
+        RecordingExporter.timerRecords(TimerIntent.TRIGGERED).getFirst().getValue();
+
+    Assertions.assertThat(timerRecord).hasDueDate(beginTime + 3000);
+
+    // when
+    beginTime = brokerRule.getClock().getCurrentTimeInMillis();
+    final BpmnModelInstance slowerModel =
+        Bpmn.createExecutableProcess("process_3")
+            .startEvent("start_4")
+            .timerWithCycle("R2/PT4S")
+            .endEvent("end_4")
+            .done();
+    testClient.deploy(slowerModel);
+
+    // then
+    assertThat(RecordingExporter.timerRecords(TimerIntent.CANCELED).getFirst()).isNotNull();
+
+    timerRecord =
+        RecordingExporter.timerRecords(TimerIntent.CREATED).limit(3).asList().get(2).getValue();
```

Comment:
This is a bit shorter:

```suggestion
        RecordingExporter.timerRecords(TimerIntent.CREATED).skip(2).getFirst().getValue();
```